src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -393,6 +441,12 @@ function getState(interaction) {
     if (response) {
         state.response = response;
     }
+
+    const $input = interaction.getContainer();
+    if ($input.hasClass('error')) {
+        state.validity = { isValid: false };
+    }
```

Comment:
Thought: is it confusing to have these 2 input classes `invalid` & `error`? (Maybe just a naming problem?)
The main difference I see is that `invalid` is only added in patternMask mode, and is being used to discard the response.

I also found inconsistent states of redness:
- a patternMask input with only `invalid` is only red when focused
- a patternMask input with `invalid error` is red all the time
- a non-patternMask input with `error` is red all the time

To tell the user to fix it, I prefer red all the time.